packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -199,28 +304,62 @@ export function ChatSettings() {
       {state === ChatSettingsState.Success && (
         <Alert severity="success">Settings saved successfully.</Alert>
       )}
+
+      {/* Language model section */}
+      <Divider>
+        <Chip label="Language model" color="primary" variant="outlined" />
+      </Divider>
       <Select
         value={inputConfig.model_provider_id}
         label="Language model"
-        onChange={e =>
+        onChange={e => {
           setInputConfig(inputConfig => ({
             ...inputConfig,
             model_provider_id: e.target.value
-          }))
-        }
+          }));
+
+          // if model is from a registry provider, show an input for the local
+          // model ID
+          const nextLmProvider = getProvider(e.target.value, lmProviders);
+          if (nextLmProvider?.registry) {
+            setShowLmLmid(true);
+            setLmLmid('*');
+          } else {
+            setShowLmLmid(false);
+          }
+        }}
         MenuProps={{ sx: { maxHeight: '50%', minHeight: 400 } }}
       >
         <MenuItem value="null">None</MenuItem>
         {lmProviders.providers.map(lmp =>
-          lmp.models
-            .filter(lm => lm !== '*') // TODO: support registry providers
-            .map(lm => (
-              <MenuItem value={`${lmp.id}:${lm}`}>
-                {lmp.name} :: {lm}
-              </MenuItem>
-            ))
+          lmp.models.map(lm => (
```

Comment:
Here as well, `lm` refers to both "Language Model" and "Local Model" — can we please use different names for both, to reduce confusion?

Reply:
Addressed in latest revision; see above.